ci/test/test-ci.py

```diff
@@ -125,12 +125,20 @@ def test_pull_request_comment_does_not_overwrite_approval():
     assert 'tpoterba' not in review_state['reviews']
 
 
-def get_pr(source_ref):
+def get_pr(source_ref,
+           delay_in_seconds=DELAY_IN_SECONDS,
+           max_polls=MAX_POLLS):
     status = ci_get('/status', status_code=200)
-    assert 'prs' in status
-    assert '_watched_targets' in status
-    all_prs = [PR.from_json(x) for x in status['prs']]
-    prs = [pr for pr in all_prs if pr.source.ref.name == source_ref]
+    polls = 0
+    prs = []
+    while len(prs) == 0:
+        assert 'prs' in status
+        assert '_watched_targets' in status
+        all_prs = [PR.from_json(x) for x in status['prs']]
+        prs = [pr for pr in all_prs if pr.source.ref.name == source_ref]
+        polls = polls + 1
+        time.sleep(delay_in_seconds)
```

Comment:
Any particular reason the delay here is placed differently than in `poll_until_pr_exists_and`. If there is some data race condition you're defending against, you may want the time.sleep as first statement in this block

Reply:
no particular reason.

I changed everything to always make a network request first, then sleep